core/src/test/scala/com/mesosphere/usi/core/logic/MesosEventsLogicTest.scala

```diff
@@ -0,0 +1,60 @@
+package com.mesosphere.usi.core.logic
+
+import com.mesosphere.mesos.client.MesosCalls
+import com.mesosphere.usi.core.helpers.FrameworkMock
+import com.mesosphere.usi.core.helpers.MesosMock
+import com.mesosphere.usi.core.matching.ScalarResourceRequirement
+import com.mesosphere.usi.core.models.PodId
+import com.mesosphere.usi.core.models.ResourceType
+import com.mesosphere.utils.UnitTest
+
+class MesosEventsLogicTest extends UnitTest {
+
+  private val mesosEventLogic = new MesosEventsLogic(new MesosCalls(FrameworkMock.mockFrameworkId))
+
+  "MesosEventsLogic" should {
+    "decline an offer when PodSpec's are empty" in {
+      val (_, schedulerEventsBuilder) = mesosEventLogic matchOffer(
+        MesosMock.mockOffer,
+        Seq()
+      )
+      schedulerEventsBuilder.result.mesosCalls.size shouldBe 1
+      val declines = schedulerEventsBuilder.result.mesosCalls.head.getDecline.getOfferIdsList
+      declines.size() shouldBe 1
+      declines.get(0) shouldEqual MesosMock.mockOffer.getId
+    }
+
+    "decline an offer when PodSpec's resource requirements are unmet" in {
+      val largeRunSpec = FrameworkMock.mockRunSpec.copy(
+        resourceRequirements = Seq(ScalarResourceRequirement(ResourceType.CPUS, Integer.MAX_VALUE))
+      )
+      val (_, schedulerEventsBuilder) = mesosEventLogic matchOffer(
+        MesosMock.mockOffer,
+        Seq(FrameworkMock.mockPodSpec.copy(runSpec = largeRunSpec))
+      )
+      schedulerEventsBuilder.result.mesosCalls.size shouldBe 1
+      val declines = schedulerEventsBuilder.result.mesosCalls.head.getDecline.getOfferIdsList
+      declines.size() shouldBe 1
+      declines.get(0) shouldEqual MesosMock.mockOffer.getId
+    }
+
+    "accept an offer when some PodSpec's resource requirements are met" in {
+      val (matchedPodIds, schedulerEventsBuilder) = mesosEventLogic matchOffer(
+        MesosMock.mockOffer,
```

Comment:
e.g. here ... so I am reading the test and I am immediately thinking - how do we know that something called mockOffer has enough resources to launch *4* pods. What if this starts to fail with `matchedPodIds.size is 2, expected 4`? What if I want to launch 6 pods and with specific resources? Do I do `mockOfferWith8CpusAnd6Gpus`? :)

Reply:
created a method to parametrize different dimensions in an offer. we can add more arguments as we needed. lmk if this is not ideal.